### PR TITLE
Clean up warnings from CSP code during build

### PR DESCRIPTION
**Pull request by AdamGlustein**

### CODE REVIEW

**Comment by robambalu** [comment]
can we enable -Werror with this change? If so, please do

File: cpp/csp/engine/Struct.h [hunk 1]
```diff
@@ -756,11 +756,15 @@ class Struct
     void decref()
     {
 //Work around GCC12 bug mis-identifying this code as use-after-free
-//#pragma GCC diagnostic push
-//#pragma GCC diagnostic ignored "-Wuse-after-free"
+#ifdef __linux__
```
Comment (svatasoiu) [inline cpp/csp/engine/Struct.h:759]: should probably be ifdef __GNUC__ since this is a gcc issue (and also the pragma is for GCC)

Comment (AdamGlustein) [reply cpp/csp/engine/Struct.h:759]: clang also defines __GNUC__ 
https://stackoverflow.com/questions/38499462/how-to-tell-clang-to-stop-pretending-to-be-other-compilers

**Comment by AdamGlustein** [comment]
> can we enable -Werror with this change? If so, please do



Made some more changes and enabled `-Werror`

**Comment by AdamGlustein** [comment]
Fixed `clang` warnings as well so this should be good to go now

File: cpp/csp/python/PyStructToJson.cpp [hunk 2]
```diff
@@ -72,7 +72,7 @@ inline rapidjson::Value toJson( const TimeDelta& val, const CspType& typ, rapidj
     // Convert TimeDelta to <sign><seconds>.<microseconds>
     // sign( 1 ) + seconds ( 18 ) + '.'( 1 ) + microseconds( 9 ) + '\0'( 1 )
     char buf[32] = {};
-    auto seconds = val.abs().asSeconds();
+    long seconds = val.abs().asSeconds();
```
Comment (robambalu) [inline cpp/csp/python/PyStructToJson.cpp:75]: curious what this warning was?

Comment (AdamGlustein) [reply cpp/csp/python/PyStructToJson.cpp:75]: https://github.com/Point72/csp/actions/runs/9896204748/job/27337875177?pr=321#step:8:2647

**Review by robambalu — APPROVED** [review]

**Pull request merged**